File: test/ts/assets/scripts/FullScreenAdScene.ts
import * as ee from '@senspark/ee-x';

const { ccclass, property } = cc._decorator;

@ccclass
export class FullScreenAdScene extends cc.Component {
    @property({ type: cc.Label, visible: true })
    private readonly _resultText: cc.Label | null = null;

    private _adsManager?: ee.IAdsManager;

    public get adsManager(): ee.IAdsManager {
        return this._adsManager;
    }

    public set adsManager(value: ee.IAdsManager | undefined) {
        this._adsManager = value;
    }

    private updateResult(result: ee.AdResult): void {
        switch (result) {
            case ee.AdResult.NotInitialized:
                this._resultText.string = "Not Initialized";
                break;
            case ee.AdResult.NotConfigured:
                this._resultText.string = "Not Configured";
                break;
            case ee.AdResult.Capped:
                this._resultText.string = "Capped";
                break;
            case ee.AdResult.NoInternet:
                this._resultText.string = "No Internet";
                break;
            case ee.AdResult.NotLoaded:
                this._resultText.string = "Not Loaded";
                break;
            case ee.AdResult.Failed:
                this._resultText.string = "Failed";
                break;
            case ee.AdResult.Canceled:
                this._resultText.string = "Canceled";
                break;
            case ee.AdResult.Completed:
                this._resultText.string = "Completed";
                break;
        }
    }

    public onShowInterstitialAdButtonPressed(): void {
        ee.Utils.noAwait(async () => {
            this._resultText.string = `---`;
            const result = await this.adsManager.interstitialAd.show();
            this.updateResult(result);
        });
    }

    public onShowRewardedAdButtonPressed(): void {
        ee.Utils.noAwait(async () => {
            this._resultText.string = `---`;
            const result = await this.adsManager.rewardedAd.show();
            this.updateResult(result);
        });
    }
}

File: test/unity/Assets/Scripts/FullScreenAdScene.cs
using UnityEngine;
using UnityEngine.UI;

namespace EETest {
    public class FullScreenAdScene : MonoBehaviour {
        [SerializeField]
        private Text _resultText;

        private EE.IAnalyticsManager _analyticsManager;

        public EE.IAdsManager AdsManager { get; set; }

        private void Awake() {
            _analyticsManager = EE.ServiceLocator.Resolve<EE.IAnalyticsManager>();
        }

        private void UpdateResult(EE.AdResult result) {
            switch (result) {
                case EE.AdResult.NotInitialized:
                    _resultText.text = "Not Initialized";
                    break;
                case EE.AdResult.NotConfigured:
                    _resultText.text = "Not Configured";
                    break;
                case EE.AdResult.Capped:
                    _resultText.text = "Capped";
                    break;
                case EE.AdResult.NoInternet:
                    _resultText.text = "No Internet";
                    break;
                case EE.AdResult.NotLoaded:
                    _resultText.text = "Not Loaded";
                    break;
                case EE.AdResult.Failed:
                    _resultText.text = "Failed";
                    break;
                case EE.AdResult.Canceled:
                    _resultText.text = "Canceled";
                    break;
                case EE.AdResult.Completed:
                    _resultText.text = "Completed";
                    break;
            }
        }

        public void OnShowAppOpenAdButtonPressed() {
            _analyticsManager.LogEvent(new ClickEvent {
                Button = "show_app_open_ad"
            });
            EE.Utils.NoAwait(async () => {
                _resultText.text = "---";
                var result = await AdsManager.AppOpenAd.Show();
                UpdateResult(result);
            });
        }

        public void OnShowInterstitialAdButtonPressed() {
            _analyticsManager.LogEvent(new ClickEvent {
                Button = "show_interstitial_ad"
            });
            EE.Utils.NoAwait(async () => {
                _resultText.text = "---";
                var result = await AdsManager.InterstitialAd.Show();
                UpdateResult(result);
            });
        }

        public void OnShowRewardedInterstitialAdButtonPressed() {
            _analyticsManager.LogEvent(new ClickEvent {
                Button = "show_rewarded_interstitial_ad"
            });
            EE.Utils.NoAwait(async () => {
                _resultText.text = "---";
                var result = await AdsManager.RewardedInterstitialAd.Show();
                UpdateResult(result);
            });
        }

        public void OnShowRewardedAdButtonPressed() {
            _analyticsManager.LogEvent(new ClickEvent {
                Button = "show_rewarded_ad"
            });
            EE.Utils.NoAwait(async () => {
                _resultText.text = "---";
                var result = await AdsManager.RewardedAd.Show();
                UpdateResult(result);
            });
        }
    }
}